packages/opentelemetry-exporter-collector/test/browser/CollectorMetricExporter.test.ts

```diff
@@ -0,0 +1,404 @@
+              metric3,
+              hrTimeToNanoseconds(metrics[2].aggregator.toPoint().timestamp)
+            );
+          }
+
+          assert.ok(
+            typeof metric4 !== 'undefined',
+            "fourth metric doesn't exist"
+          );
+          if (metric4) {
+            ensureValueRecorderIsCorrect(
+              metric4,
+              hrTimeToNanoseconds(metrics[3].aggregator.toPoint().timestamp)
+            );
+          }
+
+          const resource = json.resourceMetrics[0].resource;
+          assert.ok(typeof resource !== 'undefined', "resource doesn't exist");
+          if (resource) {
+            ensureWebResourceIsCorrect(resource);
+          }
+
+          assert.strictEqual(spyBeacon.callCount, 0);
+          ensureExportMetricsServiceRequestIsSet(json);
+
+          done();
+        });
+      });
+
+      it('should log the successful message', done => {
+        const spyLoggerDebug = sinon.stub(collectorExporter.logger, 'debug');
+        const spyLoggerError = sinon.stub(collectorExporter.logger, 'error');
+
+        collectorExporter.export(metrics, () => {});
+
+        setTimeout(() => {
+          const request = server.requests[0];
+          request.respond(200);
+
+          const response: any = spyLoggerDebug.args[1][0];
+          assert.strictEqual(response, 'xhr success');
+          assert.strictEqual(spyLoggerError.args.length, 0);
+
+          assert.strictEqual(spyBeacon.callCount, 0);
+          done();
+        });
+      });
+
+      it('should log the error message', done => {
+        const spyLoggerError = sinon.stub(collectorExporter.logger, 'error');
+
+        collectorExporter.export(metrics, () => {});
+
+        setTimeout(() => {
+          const request = server.requests[0];
+          request.respond(400);
+
+          const response1: any = spyLoggerError.args[0][0];
+          const response2: any = spyLoggerError.args[1][0];
+          assert.strictEqual(response1, 'body');
+          assert.strictEqual(response2, 'xhr error');
+
+          assert.strictEqual(spyBeacon.callCount, 0);
+          done();
+        });
+      });
+      it('should send custom headers', done => {
+        collectorExporter.export(metrics, () => {});
+
+        setTimeout(() => {
+          const request = server.requests[0];
+          request.respond(200);
+
+          assert.strictEqual(spyBeacon.callCount, 0);
+          done();
+        });
+      });
+    });
+  });
+
+  describe('export with custom headers', () => {
+    let server: any;
+    const customHeaders = {
+      foo: 'bar',
+      bar: 'baz',
+    };
+    let collectorExporterConfig: CollectorExporterConfigBrowser;
+
+    beforeEach(() => {
+      collectorExporterConfig = {
+        logger: new NoopLogger(),
+        headers: customHeaders,
+      };
+      server = sinon.fakeServer.create();
+    });
+
+    afterEach(() => {
+      server.restore();
+    });
+
+    describe('when "sendBeacon" is available', () => {
```

Comment:
the headers cannot be sent with beacon, are you sure about this test ? - they can only be sent with xhr

Reply:
a wait it is a fallback - it is sending headers using xmlhttprequest, I missed the last word, this is fine then